### PR TITLE
Cleanup formatting in error messages

### PR DESCRIPTION
**Pull request by kratman**
# Description

I noticed some of the longer f-strings printed out with large indents. This PR is to clean that up a bit. There are still some triple quoted strings that could be cleaned up, but these are the easy ones to catch.

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: src/pybamm/discretisations/discretisation.py [hunk 1]
```diff
@@ -500,8 +500,8 @@ def check_tab_conditions(self, symbol, bcs):
 
         if domain != "current collector":
             raise pybamm.ModelError(
-                f"""Boundary conditions can only be applied on the tabs in the domain
-            'current collector', but {symbol} has domain {domain}"""
+                "Boundary conditions can only be applied on the tabs in the domain"
```
Comment (kratman) [inline src/pybamm/discretisations/discretisation.py:503]:
```suggestion
                "Boundary conditions can only be applied on the tabs in the domain "
```

File: src/pybamm/expression_tree/operations/convert_to_casadi.py [hunk 2]
```diff
@@ -231,8 +231,6 @@ def _convert(self, symbol, t, y, y_dot, inputs):
 
         else:
             raise TypeError(
-                f"""
-                Cannot convert symbol of type '{type(symbol)}' to CasADi. Symbols must all be
-                'linear algebra' at this stage.
-                """
+                f"Cannot convert symbol of type '{type(symbol)}' to CasADi. Symbols must all be"
```
Comment (kratman) [inline src/pybamm/expression_tree/operations/convert_to_casadi.py:234]:
```suggestion
                f"Cannot convert symbol of type '{type(symbol)}' to CasADi. Symbols must all be "
```

**Comment by agriyakhetarpal** [comment]
Out of lazy curiosity, is there a pre-commit hook that can clean things like these?

**Review by kratman — COMMENTED** [review]
Fixing two mistakes

**Comment by kratman** [comment]
> Out of lazy curiosity, is there a pre-commit hook that can clean things like these?

Probably, these just annoyed me when I was debugging something and it only took me a couple minutes to do it. I know there are a few `"some text {} {}".replace()` and `"some text {} {}".format()` cases that are trickier to fix.

**Review by agriyakhetarpal — APPROVED** [review]

**Pull request merged**